REVIEW.md

```diff
@@ -0,0 +1,219 @@
+As with all aspects of the Code of Conduct, this applies to everyone involved in the discussion (including members of the Spyder organization on GitHub).
+
+Other notes:
+- Unless the reviewer has explicit permission on the conversation for that PR, reviewers should not commit to the same branch as the PR being reviewed.
+- If more than roughly 20% of review comments fall in the `[optional]` section described below, then it is worth adding this feedback in an issue linked to the PR instead of on the PR directly.
+
+
+## For code-focused review
+
+### Required review
+
+The following are areas of review the community expects to be covered in the review of every PR.
+Not all of the following questions will be relevant for every PR (for example a bug fix may not need to answer the same questions as a proposal for a new feature), but these are the themes to be addressed.
+
+- [ ] Motivation
+    - Do we want this?
+    - Would it help multiple users to have this accepted into Spyder?
+    - Does it help to solve bugs?
+    - Is it a necessary refactoring?
+
+- [ ] Tests
+    - Do the tests pass in the continuous integration build?
+      If appropriate, help the contributor understand why tests failed.
+    - Are regression tests needed?
+      If a reproducible bug is being fixed in this PR, it may be needed and reviewers can help the contributor design and implement them.
+      In case the contributor is unable to add the test, the reviewer should add it in a follow-up PR.
+
+ - [ ] Sustainability and maintenance
+    - Is the code easy to read and low on redundancy?
+    - Should variable names be improved for clarity or consistency?
+    - Should comments or docstrings be added?
+    - Should comments be removed as unhelpful or extraneous?
+    - Will the cost of maintaining a new feature be worth its benefits?
+    - Will the new code add any dependencies on other projects?
+
+- [ ] User-facing changes
+    - What use cases might this change influence?
+    - Does the user experience match existing patterns in Spyder?
+      If it does not, why was the choice made?
+    - Is in-interface text (new or modified) appropriately descriptive and free of spelling or grammar issues that would make its meaning unclear?
+
+Review that addresses these questions on a PR will be prefixed with `[required]` so that the PR author(s) are clear that they are blockers.
+PR authors are responsible for addressing these reviews if prompted by the reviewer in order for that PR to be eligible for merging.
+
+### Optional review
+
+The following are themes that may be helpful to cover on a PR, but are optional.
+
+**Note**: The difference between an objective improvement and a subjective preference isn’t always clear.
```

Comment:
Thinking about this phrase, should a section exists to provide guidance when a review/PR gets stuck since the PR author and reviewer don't come to a conclusion of what is the end form of the PR? (mainly in situations where author and reviewer are both core developers). Maybe something like suggesting pinging someone else to check the review and PR context or other ways to come with a conclusion of what needs to be done.

Reply:
This is pretty standard procedure (happens with me, @ccordoba12 and @dalthviz all the time) but it would be good to codify it somewhere. Since it applies to any PR, and either optional or required changes, I think it would make sense to go in a top-level section; adding it as one or more bullets in the Communication Guidelines section would probably be ideal, since several of the bullets there already touch on related issues.

Also, as a corollary, at least in my experience with core developers reviewing the PR of other core devs, there are often cases where the reviewer isn't sure of something/is asking a question, and others where the reviewer raises an issue or asks for a change that ends up not actually being applicable; in fact, I'd say that one of these ends up being case with nearly 1/2 of all the comments of core dev PRs I'm involved in either as an author or reviewer, with many if not most of those just being mistakes/misunderstandings rather than actually differences of opinion.

Everyone makes mistakes sometimes and no one is perfect, and we always have the chance to learn something new from others, no matter their level of experience. Labeling review items as "required", I think, carries with it the implication that the reviewer is correct by default, which is certainly not always the case even on new contributor PRs. As such, I'd suggest rethinking this framing a bit, which I'll address more directly in a comment on my followup review.

Reply:
Hey @isabela-pf Would you like @dalthviz or myself to propose some specific language here?

Reply:
I proposed a [suggestion](https://github.com/spyder-ide/spyder/pull/16789#discussion_r755516108) to hopefully address this; let me know what you think!